### PR TITLE
fix: update hook deps to improve ref equality checks

### PR DESCRIPTION
**Pull request by zzmp**
- Prevents re-rendering of swap info when no values have actually changed

### CODE REVIEW

File: src/lib/components/TokenImg.tsx [hunk 1]
```diff
@@ -26,16 +26,17 @@ function TokenImg({ token, ...rest }: TokenImgProps) {
...
     void attempt
 
-    const src = srcs.find((src) => !badSrcs.has(src))
-    if (!src) return <MissingToken color="secondary" {...rest} />
+    return srcs.find((src) => !badSrcs.has(src))
+  }, [attempt, srcs])
 
-    const alt = tokenInfo.name || tokenInfo.symbol
-    return <img src={src} alt={alt} key={alt} onError={onError} {...rest} />
-  }, [attempt, onError, rest, srcs, tokenInfo.name, tokenInfo.symbol])
```
Comment (zzmp) [inline src/lib/components/TokenImg.tsx:38]: rest is instantiated each time, so this `useMemo` is ineffective.

File: src/lib/hooks/swap/useSwapInfo.tsx [hunk 2]
```diff
@@ -121,13 +123,16 @@ export default function useSwapInfo(): SwapInfo {
   // swapInfo will lag behind swapAtom by a frame, because its update is triggered by swapAtom
```
Comment (ianlapham) [inline src/lib/hooks/swap/useSwapInfo.tsx:123]: do we need this comment still if context is merged? Also does that change remove need to do this

File: src/lib/hooks/useSlippage.ts [hunk 3]
```diff
@@ -27,6 +29,9 @@ export default function useSlippage(trade: InterfaceTrade<Currency, Currency, Tr
     const auto = shouldUseAutoSlippage || !maxSlippage
     const allowed = shouldUseAutoSlippage ? autoSlippage : maxSlippage ?? autoSlippage
     const warning = auto ? undefined : getSlippageWarning(allowed)
+    if (auto && allowed === DEFAULT_AUTO_SLIPPAGE) {
+      return DEFAULT_SLIPPAGE
```
Comment (ianlapham) [inline src/lib/hooks/useSlippage.ts:33]: Does this constant remove a re-render?

**Review by ianlapham — COMMENTED** [review]
Small questions

File: src/lib/hooks/swap/useSwapInfo.tsx (hunk 2 above)
Comment (zzmp) [reply src/lib/hooks/swap/useSwapInfo.tsx:123]: We will not need this comment after #3710 is merged, but #3710 removes it (and all of the logic in this hook).

File: src/lib/hooks/useSlippage.ts (hunk 3 above)
Comment (zzmp) [reply src/lib/hooks/useSlippage.ts:33]: yes - it removes a re-render on the first frame with a returned result.

File: src/lib/hooks/swap/useSwapInfo.tsx (hunk 2 above)
Comment (ianlapham) [reply src/lib/hooks/swap/useSwapInfo.tsx:123]: great

**Review by ianlapham — APPROVED** [review]
Looks good

**Pull request merged**